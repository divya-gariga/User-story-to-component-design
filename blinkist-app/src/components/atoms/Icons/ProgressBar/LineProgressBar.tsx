import React from "react";
import { theme } from "../../../../Theme/Theme"
import { LinearProgress } from "@mui/material";
import { ThemeProvider } from "@emotion/react";

interface ProgressBarProps {
  displayElement: React.CSSProperties;
  topHeight: React.CSSProperties;
  progressValue: number;
}

function LineProgressBar(props: ProgressBarProps) {
  return (
    <div >
      <ThemeProvider theme={theme}>
        <LinearProgress
          sx={{
            width: "282px",
            height: "15px",
            borderRadius: "0px 0px 8px 8px",
            display: props.displayElement,
            position: "relative",

          }}
          color="progressCompleted"
          variant="determinate"
          value={props.progressValue}
          data-testid="bar"
        />
      </ThemeProvider>
    </div>
  );
}
export default LineProgressBar;
