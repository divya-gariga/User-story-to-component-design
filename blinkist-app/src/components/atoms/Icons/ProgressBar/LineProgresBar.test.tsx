import React from 'react';
import { render, screen } from '@testing-library/react';
import LineProgressBar from './LineProgressBar';
import renderer from 'react-test-renderer';
import { LinearProgress } from '@mui/material';

test('rendering the Logo', () =>  {
render(<LineProgressBar displayElement={{display:"inline-block"}} topHeight={{height:"130px"}} progressValue={0} />)
const bar=screen.getByTestId("bar");
expect(bar).toBeInTheDocument();
});

test('snapshots',()=>{
    const renderedComponent = renderer.create(<LineProgressBar displayElement={{display:"inline-block"}} topHeight={{height:"130px"}} progressValue={0} />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
})