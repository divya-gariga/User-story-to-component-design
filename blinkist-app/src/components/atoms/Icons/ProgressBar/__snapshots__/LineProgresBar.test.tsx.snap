// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div>
  <span
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="MuiLinearProgress-root MuiLinearProgress-colorProgressCompleted MuiLinearProgress-determinate css-1bwxga9-MuiLinearProgress-root"
    data-testid="bar"
    role="progressbar"
  >
    <span
      className="MuiLinearProgress-bar MuiLinearProgress-barColorProgressCompleted MuiLinearProgress-bar1Determinate css-1i3dgut-MuiLinearProgress-bar1"
      style={
        Object {
          "transform": "translateX(-100%)",
        }
      }
    />
  </span>
</div>
`;
